server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -22,138 +24,425 @@
-                default -> throw new IllegalArgumentException("No valid expand wildcard value [" + wildcard + "]");
+        public static class Builder {
+            private boolean matchOpen = true;
+            private boolean matchClosed = false;
+            private boolean includeHidden = false;
+            private boolean resolveAliases = true;
+            private boolean allowEmptyExpressions = true;
+
+            Builder() {}
+
+            Builder(WildcardOptions options) {
+                matchOpen = options.matchOpen;
+                matchClosed = options.matchClosed;
+                includeHidden = options.includeHidden;
+                resolveAliases = options.resolveAliases;
+                allowEmptyExpressions = options.allowEmptyExpressions;
+            }
+
+            /**
+             * Open indices will be matched. Defaults to true.
+             */
+            public Builder matchOpen(boolean matchOpen) {
+                this.matchOpen = matchOpen;
+                return this;
+            }
+
+            /**
+             * Closed indices will be matched. Default to false.
+             */
+            public Builder matchClosed(boolean matchClosed) {
+                this.matchClosed = matchClosed;
+                return this;
+            }
+
+            /**
+             * Hidden indices will be included from the result. Defaults to false.
+             */
+            public Builder includeHidden(boolean includeHidden) {
+                this.includeHidden = includeHidden;
+                return this;
+            }
+
+            /**
+             * Aliases will be included in the result. Defaults to true.
+             */
+            public Builder resolveAliases(boolean resolveAliases) {
+                this.resolveAliases = resolveAliases;
+                return this;
+            }
+
+            /**
+             * If true, when any of the expressions does not match any indices, we consider the result of this expression
+             * empty; if all the expressions are empty then we have a successful but empty response.
+             * If false, we throw an error immediately, so even if other expressions would result into indices the response
+             * will contain only the error. Defaults to true.
+             */
+            public Builder allowEmptyExpressions(boolean allowEmptyExpressions) {
+                this.allowEmptyExpressions = allowEmptyExpressions;
+                return this;
+            }
+
+            /**
+             * Disables expanding wildcards.
+             */
+            public Builder none() {
+                matchOpen = false;
+                matchClosed = false;
+                includeHidden = false;
+                return this;
+            }
+
+            /**
+             * Maximises the resolution of indices, we will match open, closed and hidden targets.
+             */
+            public Builder all() {
```

Comment:
I think I slightly prefer the `none()` and `all()` methods, since they're likely to be on something with a name resembling `wildcardBuilder.none()` or `wildcardOptions.node()`, etc. But either is fine with me.

Reply:
What about the following, 
- `wildcardBuilder.matchNone()` which remains inline with the other builder methods like `matchOpen` and `matchClosed` for example.
- and `all()` remains like this because it covers `match*` and `includeHidden` builders?

Reply:
I think ultimately as long as all of the filtering methods match style wise I'd be fine with it. But also this is top tier bike shed painting so feel free to ignore this particular bit of feedback. If I were using the class as it stands today I would likely not experience any significant hurdles or cognitive dissonance with the current method names.